### DISCUSSION TITLE
Ability to modify requests

### DISCUSSION THREAD

**Discussion by niba** [olimorris/codecompanion.nvim#99] [Ideas]
Hey,

Claude introduced a new mechanism for caching requests that can significantly reduce costs. Here is the link to the article:

Blog post - https://blog.getbind.co/2024/08/15/what-is-claude-prompt-caching-how-does-it-work/
Claude caching documentation - https://docs.anthropic.com/en/docs/build-with-claude/prompt-caching

I'm wondering what is the best approach to support specific settings per AI model:

1. Give users the ability to transform request parameters, allowing them to implement everything on their own.
2. Add native support inside the plugin.

Let me know your thoughts about it. Thanks!

**Comment by olimorris** [comment]
Thanks for sharing. This is definitely the next step in the evolution of this plugin. I'll add this to the roadmap and workout how we can begin leveraging it.

**Reply by olimorris** [reply]
Please see #126 . Would love your feedback.
